api/scpca_portal/models/dataset.py

```diff
@@ -383,11 +385,32 @@ def libraries(self) -> Iterable[Library]:
     def ccdl_type(self) -> Dict:
         return ccdl_datasets.TYPES.get(self.ccdl_name, {})
 
-    @property
-    def is_data_valid(self) -> bool:
-        """Determines if the Dataset's Data attr is valid."""
-        data_validator = DataValidator(self.data)
-        return data_validator.is_valid
+    @staticmethod
+    def validate_data(data: Mapping[str, Any]) -> DatasetData:
+        return DatasetData.model_validate(data)
+
+    def validate(self) -> None:
+        """
+        Validates that projects and samples, who's ids were passed in through the data attribute,
+        both exist and are correctly related.
+        Raises exceptions if projects, samples or their associations do not exist.
+        """
+        if self.format == DatasetFormats.ANN_DATA.value:
+            if any(
+                project_data.get(Modalities.SPATIAL.value, [])
+                for project_data in self.data.values()
+            ):
+                # TODO: add custom exception
+                raise Exception("No Spatial data for ANNDATA.")
+
+        # validate that all projects exist
+        existing_ids = Project.objects.filter(scpca_id__in=self.data.keys()).values_list(
+            "scpca_id", flat=True
+        )
+        if missing_keys := set(self.data.keys()) - set(existing_ids):
+            raise Exception(f"The following projects do not exist: {list(missing_keys)}")
+
+        # TODO: sample modality existence check
```

Comment:
3 queries:
- collect all SINGLE_CELL ids that arent merged and assert that they are under the correct project based on `project__scpca_id`. I would use values list to "annotate" them. Throw error on first sample that is under wrong project.
- collect all merged and check existence of project_id and includes_merged_[format]
- collect all spatial and error on first that does not belong to project. (same as above)